helix-lsp/src/transport.rs

```diff
@@ -253,6 +253,39 @@ impl Transport {
         mut server_stdout: BufReader<ChildStdout>,
         client_tx: UnboundedSender<(usize, jsonrpc::Call)>,
     ) {
+        async fn exit_language_server(
```

Comment:
I nested this function inside of recv() because something similar is being done in send(). I can move it if that's not a pattern that you want to be copied.